lib/explorer/series.ex

```diff
@@ -476,6 +476,41 @@ defmodule Explorer.Series do
   @spec last(series :: Series.t()) :: any()
   def last(series), do: Shared.apply_impl(series, :last, [])
 
+  @doc """
+  Returns a series from two series, based on a predicate.
+
+  The resulting series is built by evaluating each element of
+  `predicate` and returning either the corresponding element from
+  `on_true` or `on_false`.
+
+  `predicate` must be a boolean series. `on_true` and `on_false` must be
+  a series of the same length as `pred`.
+  """
+  @doc type: :transformation
+  @spec select(predicate :: Series.t(), on_true :: Series.t(), on_false :: Series.t()) :: Series.t()
+  def select(
+        predicate,
+        %Series{dtype: on_true_dtype} = on_true,
+        %Series{dtype: on_false_dtype} = on_false
+      )
+      when K.and(numeric_dtype?(on_true_dtype), numeric_dtype?(on_false_dtype)),
+      do: Shared.apply_impl(predicate, :select, [on_true, on_false])
+
+  def select(
+        predicate,
+        %Series{dtype: on_true_dtype} = on_true,
+        %Series{dtype: on_false_dtype} = on_false
+      )
+      when on_true_dtype == on_false_dtype,
+      do: Shared.apply_impl(predicate, :select, [on_true, on_false])
```

Comment:
We should check on both cases that `predicate` is a boolean. :)

Reply:
Great point, will do